src/main/java/com/mcsunnyside/coreprotecttnt/Main.java

```diff
@@ -74,6 +76,52 @@ public void run() {
         }.runTaskTimerAsynchronously(this, 0, 20 * 60);
     }
 
+    @EventHandler
+    public void onPlayerInteract(PlayerInteractEvent e) {
+        if (!getConfig().getBoolean("bed.log")) {
+            return;
+        }
+
+        Block bed = e.getClickedBlock();
+
+        if (e.getAction() != Action.RIGHT_CLICK_BLOCK ||
+                bed.getLocation().getWorld().getEnvironment() == World.Environment.NORMAL) {
+            return;
+        }
+
+        if (!(bed.getBlockData() instanceof Bed)) {
+            return;
+        }
+
+        Bed data = (Bed) bed.getBlockData();
+        Location location = bed.getLocation();
+        if(data.getPart() == Bed.Part.FOOT) {
+            location = location.add(data.getFacing().getDirection());
```

Comment:
Location#add, Location#subtract and other methods mutate the Location and don't create a new Location, so you don't need `location = ` part.